homeassistant/components/cloud/helpers.py

```diff
@@ -29,3 +43,237 @@ def _get_logs() -> list[str]:
             return [self.format(record) for record in records]
 
         return await hass.async_add_executor_job(_get_logs)
+
+
+class LLMChatHelper:
+    """Helper methods for LLM chat handling."""
+
+    @staticmethod
+    async def prepare_chat_for_generation(
+        hass: HomeAssistant,
+        chat_log: conversation.ChatLog,
+        response_format: dict[str, Any] | None = None,
+    ) -> dict[str, Any]:
+        """Prepare kwargs for Cloud LLM from the chat log."""
+
+        messages = [
+            message
+            for content in chat_log.content
+            if (message := LLMChatHelper._convert_content_to_chat_message(content))
+        ]
+
+        if not messages or messages[-1]["role"] != "user":
+            raise HomeAssistantError("No user prompt found")
+
+        last_content = chat_log.content[-1]
+        if last_content.role == "user" and last_content.attachments:
+            files = await LLMChatHelper.async_prepare_files_for_prompt(
+                hass, last_content.attachments
+            )
+            user_message = messages[-1]
+            current_content = user_message.get("content", [])
+            user_message["content"] = [*(current_content or []), *files]
+
+        tools: list[ToolParam] = []
+        tool_choice: str | None = None
+
+        if chat_log.llm_api:
+            ha_tools: list[ToolParam] = [
+                LLMChatHelper._format_tool(tool, chat_log.llm_api.custom_serializer)
+                for tool in chat_log.llm_api.tools
+            ]
+
+            if ha_tools:
+                if not chat_log.unresponded_tool_results:
+                    tools = ha_tools
+                    tool_choice = "auto"
+                else:
+                    tools = []
+                    tool_choice = "none"
+
+        web_search = WebSearchToolParam(
```

Comment:
Will web search always be available? Or is that something we can enable/disable?

Reply:
We can configure it in the backend but it can be always overriden by the SDK when performing requests. Why wouldn't we want to always have web search? 
Do you think this is something the user should be able to toggle?